github-runner-manager/tests/unit/test_pre_job_script.py

```diff
@@ -465,6 +468,95 @@ def test_allow_external_contributor_enabled_skips_check(
     assert LOG_CHECK_PASSED not in result.stderr
 
 
+@pytest.mark.parametrize(
+    "author_association,should_pass",
+    [
+        ("OWNER", True),
+        ("MEMBER", True),
+        ("COLLABORATOR", True),
+        ("CONTRIBUTOR", True),  # Allowed for private repos
+        ("FIRST_TIME_CONTRIBUTOR", False),  # Still blocked
+        ("NONE", False),  # Still blocked
+    ],
+)
+def test_private_repository_allows_contributor(
+    pre_job_template: Template,
+    github_env_vars: Dict[str, str],
+    tmp_path: Path,
+    author_association: str,
+    should_pass: bool,
+    default_template_vars: Dict,
+):
+    """Test that private repositories extend allowed authorizations to include CONTRIBUTOR.
+
+    Private repositories trust CONTRIBUTORs since they have repository access,
+    but still block FIRST_TIME_CONTRIBUTOR and NONE.
```

Comment:
Can you change the doc string to the "arrange act assert" pattern we usually use?